src/jit/assertionprop.cpp

```diff
@@ -2373,6 +2381,15 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* stmt, Gen
     ValueNum vnCns = tree->gtVNPair.GetConservative();
     ValueNum vnLib = tree->gtVNPair.GetLiberal();
 
+    // We typically want to use the Normal ValueNumber when checking for constants
+    // except for the nodes that might generate an exception like GT_DIV, GT_MOD, etc...
+    //
+    if (!tree->OperMayThrow(this))
```

Comment:
So I guess this is needed because if we have a VN exception set we don't really know if it comes from the node itself or from its children?